Assets/Scripts/Pathfinding/Pathfinding.cs

```diff
@@ -96,9 +99,20 @@ private static List<Edge> DetermineFoundPath(this EndPoint self, IVertex end)
             {
                 vertexPath.AddFirst(tempEnd);
             }
+            vertexPath.AddFirst(self);
 
             // return the edges connecting the vertices in the path
-            return vertexPath.Zip(vertexPath.Skip(1), (v1, v2) => v1.GetEdge(v2)).ToList();
+            var path = vertexPath.ZipThree(
+                vertexPath.Skip(1),
+                vertexPath.Skip(2),
+                (v1, v2, v3) =>
+                    new RouteSegment(track: v1.GetEdge(v2), laneType: v2.SubRoute(v2.GetEdge(v1), v2.GetEdge(v3)))
```

Comment:
Wieso wird bei der SubRoute die Edge von v2 zu v1 genutzt und nicht anders herum?

Reply:
Weil in v2 nur die Edge von v2 zu v1 gespeichert ist. Die von v1 nach v2 ist ja ein anderes Objekt.

Reply:
Aber das ändert doch nicht, dass wir die Edge haben wollen, die zur Vertex führt. Und das ist doch v1 zu v2, oder?

Reply:
In subroute geht es bloß um den benötigten LaneType um von der ersten Edge zur zweiten Edge zu kommen. Die Richtung der Edge ist dabei egal, wichtig ist nur, dass die Edges in der Section mit Namen (Lagebeziehung) gespeichert sind. Der eigentliche Track ist `v1.GetEdge(v2)`, also die edge von v1 nach v2.

Reply:
Ja, war nen Denkfehler.
Vielen Dank für die aufschlussreiche Antwort und produktive Diskussion.

Reply:
Ist denke ich nen bisschen unintuitiv, dass SubRoute so implementiert ist.
Wäre dafür, dass SubRoute bei seiner ersten Edge erstmal `.other` aufruft.
Wenn man nämlich die SubRoute Funktion sieht, denkt man dass sie einem sagt, wie man abbiegen muss, wenn man auf edge1 steht und zu edge2 möchte.

Reply:
e4c03cd